tests/unit/test_models_transformers_question_answering.py

```diff
@@ -12,12 +12,22 @@
 )
 
 import torch
+from tempfile import TemporaryDirectory
 
 NUM_GPUS = max(1, torch.cuda.device_count())
 BATCH_SIZE = 8
 
 
-@pytest.fixture()
+@pytest.fixture(scope="module")
```

Comment:
Don't understand very well what you want to do here. There is another fixture called tmp in configtest https://github.com/microsoft/nlp-recipes/blob/staging/tests/conftest.py#L93. The only difference is that the one in conftest has the scope function (i.e. creates a temp folder for each test function that is executed) and the one here has scope module (i.e.  creates a temp folder for each test file, several test functions). 

There is a name conflict, it is interesting that pytest is not raising an error. It has to be taking one of them, but not sure which one. 

I think the safest thing would be to just erase this one and use the one in conftest with scope function, that way you are creating a folder each time. Unless you have in mind a different thing

Reply:
I need a tmp directory with a module scope because I'm using it within another fixture, qa_test_data. 
When the tmp directory is function scoped, it gets deleted after the qa_test_data fixture is instantiated and I got an error in the actual test saying the tmp directory couldn't be found.
About the name conflict, pytest will look for tmp  within the current script first. If it couldn't find it, it will look for it in confest. See details here https://docs.pytest.org/en/latest/fixture.html

Reply:
ok got it, makes sense, in that case I would suggest to state that explicitly. For example by writing a comment saying that ` name conflict, pytest will look for tmp within the current script first. If it couldn't find it, it will look for it in confest`. Other example would be to change the name to `tmp_scope` or something. Or even consider to move it to conftest in case it is used by others. Whatever you want

Reply:
Sounds good. I think it's a good idea to create a tmp_module in conftest. Changes coming soon.